app/models/openscap_result.rb

```diff
@@ -0,0 +1,76 @@
+class OpenscapResult < ApplicationRecord
+  belongs_to :container_image
+  has_one    :binary_blob,           :dependent => :destroy, :autosave => true, :as => :resource
+  has_many   :openscap_rule_results, :dependent => :destroy, :autosave => true
+
+  def self.openscap_available?
+    # needed only for travis
+    require 'openscap'
+    require 'openscap/ds/arf'
+    require 'openscap/xccdf/benchmark'
+    true
+  rescue LoadError
+    false
+  end
+
+  def raw=(value)
+    raise "Cannot invoke raw more than once." unless binary_blob.nil?
```

Comment:
1. can also be added to the association `has_one :binary_blob, required: true` 
2. makes sense, if `OpenscapResult` is created only once and probably never updated. Otherwise I would add some logic to check if the blob has changed. But having it always create the rules once the blob changes makes sense
3. I think 1&2 remove the need for `raw=` 

ad 3. you might want to have a similar method, named `attach_raw_result(openscap_xml)` which hides the BinaryBlob